ZCRMiOS/Classes/DashBoardAPIHandler.swift

```diff
@@ -0,0 +1,1737 @@
+            {
+                OnCompletion(nil,error)
+                print("APIResponse is NIL")
+                return
+            }
+            
+            OnCompletion(APIResponse,nil)
+            
+        }
+        
+        
+    } // func ends
+    
+    
+    internal func refreshDashBoardWith(ID dbID: Int64, OnCompletion: @escaping refreshResponse) {
+        
+        let URLPath = "\(SLASH)\(URLPathName.ANALYTICS)\(SLASH)\(dbID)\(SLASH)\(URLPathName.REFRESH)"
+        
+        setUrlPath(urlPath: URLPath)
+        
+        setRequestMethod(requestMethod: .POST)
+        
+        setJSONRootKey(key: JSONRootKey.DATA)
+        
+        let request = APIRequest(handler: self)
+        request.getAPIResponse {  ( APIResponse, error ) in
+            
+            guard error == nil else {OnCompletion(APIResponse,error); return}
+            guard let APIResponse = APIResponse else
+            {
+                OnCompletion(nil,error)
+                print("APIResponse is NIL")
+                return
+            }
+            
+            OnCompletion(APIResponse,nil)
+            
+        }
+        
+    } // func ends
+    
+    
+    internal func getDashBoardComponentColorThemes(OnCompletion: @escaping ArrayOfColorThemes) {
+        
+        let URLPath = "\(SLASH)\(URLPathName.ANALYTICS)\(SLASH)\(URLPathName.COLORTHEMES)"
+        
+        print("URL Path \(URLPath)")
+        
+        setUrlPath(urlPath: URLPath)
+        
+        setRequestMethod(requestMethod: .GET)
+        
+        setJSONRootKey(key: JSONRootKey.NILL)
+        
+        let request = APIRequest(handler: self)
+        request.getAPIResponse {  ( APIResponse, error ) in
+            
+            guard error == nil else {OnCompletion(nil,APIResponse,error); return}
+            guard let APIResponse = APIResponse else
+            {
+                OnCompletion(nil,nil,error)
+                print("APIResponse is NIL")
+                return
+            }
+            
+            let colorThemesResponseJSON = APIResponse.getResponseJSON() //[String:Any]
+            
+            let colorThemesJSON = colorThemesResponseJSON.getArrayOfDictionaries(key: colorPaletteAPINames.colorThemes)
+            
+            
+            let ArrayOfcolorThemes = self.getArrayOfColorThemesObjFrom(colorThemesJSON)
+            
+            OnCompletion(ArrayOfcolorThemes,APIResponse,nil)
+            
+        }
+        
+    } // func ends
+    
+    
+} // extension ends
+
+
+//MARK:- HANDLER PARSING FUNCTIONS
+//MARK:-
+
+extension DashBoardAPIHandler {
+    
+    //MARK: *** DASHBOARD PARSERS ***
+    
+    internal func getDashBoardObjectFrom(_ dashBoardJSON: [String:Any] ) -> ZCRMDashBoard {
+        
+        let dashBoardObj = ZCRMDashBoard()
+        
+        for (key,value) in dashBoardJSON {
+            
+            switch (key) {
+                
+            case dashBoardAPINames.dashBoardID :
```

Comment:
Why didn't you use our dictionary extension?

Reply:
Anna if switch case is used ,  we can find out if any other KEYS other than the ones which we are expecting has occurred in JSON . This will be printed as a debug message by the DEFAULT CASE .

Reply:
https://github.com/zoho/CRM-iOSSDK/pull/15#discussion_r214515160